rules/mapping/bowtie.rules

```diff
@@ -32,7 +32,17 @@ rule bowtie2_align:
     params:
         genome="{genome}",
         index_fp=str(Cfg['mapping']['genomes_fp'])
-    shell: "bowtie2 --threads {threads} -x {params.index_fp}/{params.genome} -1 {input.rp[0]} -2 {input.rp[1]} -S {output}"
+    run:
+        sample = Samples[wildcards.sample]
+        # Assume fastq input unless input filename ends with an A
+        fmt_arg = "-q"
+        if input.rp[0].endswith(("a", "A")):
+            fmt_arg = "-f"
+        # Assume unpaired, unless the paired attribute is True for the sample
```

Comment:
That's right, I only thought of it because I was trying to feed contigs back in as input samples and realized I had it assuming paired fastq's.  I'll add a test to cover it too.